js/store.js

```diff
@@ -29,31 +44,39 @@ const store = {
     return null;
   },
 
-  setFirstParty(param) {
-    /**
-     * @param object with hostname as key
-     * @todo validate the param.-
-     * @todo code to be updated in next PR
-     */
-    return this.set(param);
+  setFirstParty(hostname, data) {
+    if (!hostname) {
+      throw new Error('setFirstParty requires a valid hostname argument');
+    }
+
+    const websites = clone(this._websites);
+    websites[hostname] = data;
+    this._write(websites);
   },
 
-  async set(websites) {
-    /**
-      * @todo code to be updated in next PR
-    */
-    return await browser.storage.local.set({ websites });
+  setThirdParty(origin, target, data) {
```

Comment:
@biancadanforth @jonathanKingston here we could use `document` instead of `origin`, if the data comes from document and not origin.